apps/builder/app/builder/features/style-panel/shared/css-value-input/parse-intermediate-or-invalid-value.ts

```diff
@@ -44,6 +44,13 @@ export const parseIntermediateOrInvalidValue = (
   }
 
   if ("unit" in styleValue && styleValue.unit === "number") {
+    // when unit is number some properties supports only integer
+    // for example z-index
+    styleInput = parseCssValue(property, `${Math.round(Number(value))}`);
```

Comment:
I think you can only round it for some properties like z-index, but not generally, as you may want to have 1,5rem or something